src/odemis/driver/test/npmc_test.py

```diff
@@ -235,6 +230,30 @@ def test_multimove(self):
         f2.result()
         test.assert_pos_almost_equal(orig_pos, self.dev.position.value, **COMP_ARGS)
 
+    def test_reference(self):
+        """
+        Test referencing
+        """
+
+        # Test cancellation of referencing
+        self.dev.moveAbs({'x': 0.5}).result()
+        f = self.dev.reference({'x'})
+        time.sleep(0.01)
+        f.cancel()
+        self.assertFalse(self.dev.referenced.value['x'])
+        self.assertNotEqual(self.dev.position.value['x'], 0)
+
+        # NOTE!!! If homing aborts, you MUST run referencing again. Otherwise,
+        # the controller will be stuck and won't do anything!
+
+        # Test proper referencing
+        self.assertFalse(self.dev.referenced.value['x'])
+        self.dev.reference({'x'}).result()
+        self.assertTrue(self.dev.referenced.value['x'])
+        # The new position shoudl be the origin (0)
+        origin = {'x': 0}
```

Comment:
Actually, it should report the POS_COR. But, to keep it simple, you could just put POS_COR=0 before the beginning of the test.